eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/PayloadBasedTagResolver.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventMessage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Simple implementation of the {@link TagResolver} that is able to combine several lambdas to construct
+ * {@link Tag Tags} from payloads of type {@code P}.
+ * <p>
+ * The results of invoking all configured lambdas are combined into the {@link Set} of {@code Tags} when invoking
+ * {@link #resolve(EventMessage)}.
+ *
+ * @param <P> The payload for which to resolve a {@link Set} of {@link Tag Tags} for.
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class PayloadBasedTagResolver<P> implements TagResolver<EventMessage<P>> {
```

Comment:
Shouldn't the PayloadTagResolver implement TagResolver<EventMessage<?>>. Effectively, it will not return any tags for events with other payloads, but any component that expects a PayloadResolver<EventMessage<?>> should be able to deal with an instance of PayloadBasedTagResolver, right?